### PR TITLE
Add new `Minitest/AssertionInHook` cop

### PR DESCRIPTION
**Pull request by fatkodima**
This cop checks for usage of assertions in hooks. 
Hooks should only be used for preparing/cleaning before/after test run.

### CODE REVIEW

File: lib/rubocop/cop/minitest/assertion_in_hook.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      #     end
+      #   end
+      #
+      class AssertionInHook < Cop
+        include MinitestExplorationHelpers
+
+        MSG = 'Do not use `%<assertion>s` in `%<hook>s` hook.'
+
+        def on_class(class_node)
+          return unless minitest_test_subclass?(class_node)
```
Comment (koic) [inline lib/rubocop/cop/minitest/assertion_in_hook.rb:29]: I'm worried that only subclasses that directly extend `Minitest::Test` will be detected. Therefore I've open 
 https://github.com/rubocop-hq/rubocop-minitest/pull/99. What do you think about it?

**Comment by fatkodima** [comment]
Updated.

File: lib/rubocop/cop/minitest/assertion_in_hook.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Minitest
+      # This cop checks for usage of assertions in hooks.
```
Comment (koic) [inline lib/rubocop/cop/minitest/assertion_in_hook.rb:6]:
```suggestion
      # This cop checks for usage of assertions in lifecycle hooks.
```

File: lib/rubocop/cop/minitest/assertion_in_hook.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      #   end
+      #
+      #   # good
+      #   class FooTest < Minitest::Test
+      #     def test_something
+      #       assert_equal(foo, bar)
+      #     end
+      #   end
+      #
+      class AssertionInHook < Cop
```
Comment (koic) [inline lib/rubocop/cop/minitest/assertion_in_hook.rb:23]: Can you rename to `AssertionInLifecycleHook`?

```suggestion
      class AssertionInLifecycleHook < Cop
```

**Comment by fatkodima** [comment]
Updated.

File: CHANGELOG.md [hunk 4]
```diff
@@ -14,6 +15,7 @@
 * [#84](https://github.com/rubocop-hq/rubocop-minitest/pull/84): New cops `AssertKindOf` and `RefuteKindOf` check for use of `assert_kind_of`/`refute_kind_of` instead of `assert(foo.kind_of?(Class))`/`refute(foo.kind_of?(Class))`. ([@fatkodima][])
 * [#85](https://github.com/rubocop-hq/rubocop-minitest/pull/85): Add autocorrect to `Rails/AssertEmptyLiteral` cop. ([@fatkodima][])
 
+
```
Comment (koic) [inline CHANGELOG.md:18]: This line seems redundant :-)

Comment (fatkodima) [reply CHANGELOG.md]: ouch. fixed.

**Comment by koic** [comment]
Thanks!

**Pull request merged**